### ISSUE TITLE
Ocarina of time - Some part of graphic packs are not loading

### ISSUE THREAD

**Issue by LuismaSP89** [gonetz/GLideN64#1985]
Note: Using this pack: http://www.emutalk.net/threads/55307-Zelda-Ocarina-of-time-Community-Retexture-Project-V7

![sin titulo](https://user-images.githubusercontent.com/12205048/51150816-7891d780-1868-11e9-98b7-9e83363af0eb.jpg)
![sin titulo2](https://user-images.githubusercontent.com/12205048/51150817-7891d780-1868-11e9-9ccc-83ba64ccbd31.jpg)
![sin titulo3](https://user-images.githubusercontent.com/12205048/51150818-7891d780-1868-11e9-879f-e455bd1244c3.jpg)
![sin titulo4](https://user-images.githubusercontent.com/12205048/51150819-792a6e00-1868-11e9-98f5-b54d788b654f.jpg)

The problem here, is that some textures are not loading until you do some thing in order to trigger it. For example, as you can see, in the 1st and 2nd picture, the texture of his cape are not loaded. You need to hit with your sword in the battle in order to trigger the loading of the texture 3rd and 4th images.

This also happens in the sky and grass textures. When you´re outside, in the hyrule field, the original textures shows, and you need to walk some distance in order to load the HD textures. (But this is a different case, since at least the original textures are loaded, because the texture cap in the 1st and 2nd images are not the original)

I don´t know if this is related with the 999Mb limit of the enhanced texture cache size. Maybe rising this limit the texture pack can be loaded entirely, but I don´t know if this can be solved this way.

**Comment by Frank-74** [comment]
@LuismaSP89 Shouldn't be a limit of 999Mb (1024 really), as latest Project64 is now LargeAddressAware (https://github.com/project64/project64/commit/63b4a4c00373a57cebc51618dcec2f8b30d91e79).  No longer needs patching for large texture packs.  Try latest nightly build from pj64 site.

**Comment by LuismaSP89** [comment]
@Frank-74 I´m using one of the latest project64 builds (2.4.0.849-gff8bb9d3) but even if project64 have a LargeAddressAware, Gliden64 only can manage 999Mb. (You can see when the texture pack is loaded, it cuts and launches the game at 999Mb)

The sky box on intro is the same when you go to the hyrule field, you need to walk some distance in order to "trigger" the texture pack of sky. The same thing happens with other textures such as the Ganondorf´s cap (You need to hit with your sword)

**Comment by LuismaSP89** [comment]
Sure, here you go:

[The Legend of Zelda - Ocarina of Time (U) (V1.0).pj.zip](https://github.com/gonetz/GLideN64/files/2761346/The.Legend.of.Zelda.-.Ocarina.of.Time.U.V1.0.pj.zip)

His cap is gray, you need to hit Ganondorf with your sword in order to load the new texture and turn his cape red.

**Comment by Frank-74** [comment]
Yes, same happens here with his cape.

Save was a bit weird on loading, maybe because of language difference?

On loading savestate:
![gliden64_the_legend_of_zelda_003](https://user-images.githubusercontent.com/11056843/51208838-a594c780-1906-11e9-9d5e-144793a0e45f.png)

After pressing START twice:
![gliden64_the_legend_of_zelda_004](https://user-images.githubusercontent.com/11056843/51208875-bfcea580-1906-11e9-8520-ed1307295960.png)

**Comment by LuismaSP89** [comment]
Nope, it loads the same for me, that colors dissapear after exit the start screen. I think is related with the savestates method of emulator.

Here´s another savestate done with the latest version of the link you provided, and is all the same:

[The Legend of Zelda - Ocarina of Time (U) (V1.0).pj.zip](https://github.com/gonetz/GLideN64/files/2761405/The.Legend.of.Zelda.-.Ocarina.of.Time.U.V1.0.pj.zip)

**Comment by LuismaSP89** [comment]
Yes, return is regresar.

**Comment by Frank-74** [comment]
Weird that all the buttons and language changed to English after pressing START twice, but Regresar remained.

**Comment by LuismaSP89** [comment]
Yes, weird indeed... xD

Btw, so, you can confirm the cape problem? Like I said, it happens for various textures in this game. And I think the problem is that, only 999Mb of memory to load a texture pack. Then, there are some texture that are not "included" in the cache, and only are loading when certain events are triggered.

**Comment by Frank-74** [comment]
If GLideN64 sets limit to 999, then it should be increased to 4096Mb.

**Comment by Frank-74** [comment]
Enhanced texture size in GLideN64 only goes upto 999Mb, but I had it set to 100Mb and I have cache of 959 MB (1,006,051,666 bytes).  So I don't think the 999 limit in gui works.

**Comment by LuismaSP89** [comment]
Then we must wait to Gonetz to clarify it. 

Btw, there´s no relation between the extracted size of the pack (806 MB) and the RAM consumed when it´s loaded, (At least is not exactly the same) you can see this for example in Dolphin´s emulator. You can use a texture pack of 500MB and when you load the textures you´ll need 2GB of RAM only for that textures. I don´t know how to explain it technically.

**Comment by gonetz** [comment]
I would test x64 version of GLideN64 for mupen64plus to be sure that it is memory allocation issue.
VRAM size is not so important here, because HD textures loaded into VRAM on demand. You just need enough RAM to keep loaded pack in it.

HD texture also may not load because plugin calculates checksum of original N64 texture incorrectly and thus can't find its HD counterpart. I fixed most of such issues, but some still may remain.
